fabric-key-binding-api-v1/src/main/java/net/fabricmc/fabric/api/client/keybinding/v1/FabricKeyBinding.java

```diff
@@ -0,0 +1,204 @@
+	 *
+	 * @return configured KeyCode
+	 */
+	public InputUtil.KeyCode getBoundKey() {
+		return getBoundKeyOf(this);
+	}
+
+	/**
+	 * Returns the configured KeyCode assigned to the KeyBinding from the player's settings.
+	 *
+	 * @param keyBinding the keybinding
+	 * @return configured KeyCode
+	 */
+	public static InputUtil.KeyCode getBoundKeyOf(KeyBinding keyBinding) {
+		return ((KeyCodeAccessor) keyBinding).getKeyCode();
+	}
+
+	public static class Builder {
+		private static final int UNASSIGNED = InputUtil.UNKNOWN_KEYCODE.getKeyCode();
+
+		private InputUtil.Type type = InputUtil.Type.KEYSYM;
+		private Identifier id = null;
+		private String translationKey;
+		private boolean unassigned = false;
+		private BooleanSupplier toggleFlagSupplier = null;
+		private int code = UNASSIGNED;
+		private String category = KeyCategories.MISC;
+
+		private Builder() {
+		}
+
+		/**
+		 * Sets the key binding id and translation key for this builder.
+		 * <br>
+		 * Key bindings will be assigned a translation key of the format "key.{namespace}.{path}"
+		 *
+		 * @param id Unique identifier for the bound key.
+		 */
+		public Builder id(Identifier id) {
+			this.id = Objects.requireNonNull(id, "Keybinding's id can not be null!");
+			this.translationKey = String.format("key.%s.%s", id.getNamespace(), id.getPath());
+			return this;
+		}
+
+		/**
+		 * Sets the translation key for the key's category. {@link KeyCategories} for
+		 * all possible values, vanilla values.
+		 *
+		 * @param category The category under which key bindings created by this builder will be grouped.
+		 */
+		public Builder category(String category) {
+			this.category = Objects.requireNonNull(category, "Keybinding's category can not be null!");
+			return this;
+		}
+
+		/**
+		 * Sets the default key to be used for the key binding created using this builder.
+		 *
+		 * @param keyCode The default key code. Must be a valid key. May not be -1.
+		 */
+		public Builder code(int keyCode) {
+			Preconditions.checkState(keyCode != UNASSIGNED, "UNASSIGNED is not a valid key code.");
+			this.code = keyCode;
+			this.unassigned = false;
+			return this;
+		}
+
+		/**
+		 * Indicates to this builder that keybindings built through it are intended to be unbound.
+		 */
+		public Builder unassigned() {
+			this.code = UNASSIGNED;
+			this.unassigned = true;
+			return this;
+		}
+
+		/**
+		 * Sets this builder to create sticky keybindings that will toggle their state when pressed.
+		 */
+		public Builder sticky() {
+			return sticky(() -> true);
+		}
+
+		/**
+		 * Sets a sticking action to be used by the constructed key binding which can be used to switch between
+		 * a sticky (toggle) and a regular key binding.
+		 *
+		 * @param toggleFlagSupplier A getter function to determine whether to toggle or not. True for toggling behaviour, false otherwise.
+		 */
+		public Builder sticky(BooleanSupplier toggleFlagSupplier) {
```

Comment:
I'd use a little nested class instead of BooleanSupplier that can then use a better method name and document it.